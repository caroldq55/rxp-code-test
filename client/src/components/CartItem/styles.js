import styled from "styled-components";

const CartItemContainer = styled.div`
  margin: 5px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const ItemName = styled.span`
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
`;

export { CartItemContainer, ItemName };
