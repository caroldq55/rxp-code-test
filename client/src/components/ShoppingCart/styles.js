import styled from 'styled-components';

const ShopingCartContainer = styled.div`
  margin: 5px;
  border: 0.0625em solid #dcdfe1;
  border-radius: 0.25em;
  min-height: 50px;
`;

const Heading = styled.div`
  padding: 5px 10px;
  background-color: #ffd633;
  font-weight: bold;
`;

const Content = styled.div`
  padding: 10px;
`;

const TotalPrice = styled.div`
  font-weight: bold;
  margin-top: 10px;
`;

export { ShopingCartContainer, Heading, Content, TotalPrice };
