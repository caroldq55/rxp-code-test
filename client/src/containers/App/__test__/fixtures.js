import { fromJS } from 'immutable';

const products = [
  {
    isPublished: 'true',
    productName: 'Apple iPhone X',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-x/iphone-x-silver-grid.png',
    price: '299'
  },
  {
    isPublished: 'true',
    productName: 'Apple iPhone 8',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-8/iphone-8-silver-grid.png',
    price: '100'
  },
  {
    isPublished: 'false',
    productName: 'Apple iPhone 8 Plus',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-8/iphone-8plus-space-grey-grid.png',
    price: '99'
  },
  {
    isPublished: 'true',
    productName: 'Sony Xperia XA2',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/sony-xperia-xa2/sony_xperia_xa2_front_v1.png',
    price: '19'
  }
];

const item = {
  isPublished: true,
  productName: 'Sony Xperia XA2',
  productImage:
    'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/sony-xperia-xa2/sony_xperia_xa2_front_v1.png',
  price: '19'
};

const cartItems = [
  {
    isPublished: true,
    productName: 'Sony Xperia XA2',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/sony-xperia-xa2/sony_xperia_xa2_front_v1.png',
    price: '19'
  }
];

const publishedProducts = [
  {
    isPublished: true,
    productName: 'Apple iPhone X',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-x/iphone-x-silver-grid.png',
    price: '299'
  },
  {
    isPublished: true,
    productName: 'Apple iPhone 8',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-8/iphone-8-silver-grid.png',
    price: '100'
  },
  {
    isPublished: true,
    productName: 'Sony Xperia XA2',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/sony-xperia-xa2/sony_xperia_xa2_front_v1.png',
    price: '19'
  }
];

const availableProducts = [
  {
    isPublished: true,
    productName: 'Apple iPhone X',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-x/iphone-x-silver-grid.png',
    price: '299'
  },
  {
    isPublished: true,
    productName: 'Apple iPhone 8',
    productImage:
      'https://www.telstra.com.au/content/dam/tcom/personal/mobile-phones/product-catalogue/iphone-8/iphone-8-silver-grid.png',
    price: '100'
  }
];

const initialValue = fromJS({
  products: [],
  loading: false,
  cartItems: []
});

export {
  products,
  initialValue,
  cartItems,
  availableProducts,
  item,
  publishedProducts
};
