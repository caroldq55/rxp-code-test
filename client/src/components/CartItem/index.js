import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

import { CartItemContainer, ItemName } from './styles';

const CartItem = ({ item, handleRemoveCartItem }) => (
  <CartItemContainer>
    <ItemName>{item.productName}</ItemName>
    <Button
      variant='warning'
      onClick={() => {
        handleRemoveCartItem(item);
      }}
      data-test-id='removeButton'
    >
      Remove
    </Button>
  </CartItemContainer>
);

CartItem.propTypes = {
  item: PropTypes.shape({
    productImage: PropTypes.string.isRequired,
    productName: PropTypes.string.isRequired,
    price: PropTypes.string.isRequired,
    isPublished: PropTypes.bool
  }).isRequired,
  handleRemoveCartItem: PropTypes.func.isRequired
};

export default CartItem;
