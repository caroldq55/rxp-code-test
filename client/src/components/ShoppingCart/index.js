import { isEmpty } from 'lodash';
import React from 'react';
import PropTypes, { arrayOf } from 'prop-types';

import CartItem from '../CartItem';
import { ShopingCartContainer, Heading, Content } from './styles';

const ShoppingCart = ({ cartItems, handleRemoveCartItem }) => (
  <ShopingCartContainer>
    <Heading>Shopping Cart</Heading>
    {isEmpty(cartItems) ? (
      <Content>Your cart is empty</Content>
    ) : (
      <Content>
        {cartItems.map((item, index) => (
          <CartItem
            key={index}
            item={item}
            handleRemoveCartItem={handleRemoveCartItem}
          />
        ))}
      </Content>
    )}
  </ShopingCartContainer>
);

ShoppingCart.propTypes = {
  cartItems: arrayOf(
    PropTypes.shape({
      productName: PropTypes.string.isRequired
    })
  ),
  handleRemoveCartItem: PropTypes.func.isRequired
};

ShoppingCart.defaultProps = {
  cartItems: []
};

export default ShoppingCart;
