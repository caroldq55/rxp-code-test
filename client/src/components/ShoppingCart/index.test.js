import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import ShoppingCart from '.';
import CartItem from '../CartItem';
import { cartItems } from '../../containers/App/__test__/fixtures';

CartItem.displayName = 'CartItem';

let wrapper;
const handleRemoveCartItemSpy = jest.fn();

describe('<ShoppingCart/>', () => {
  beforeEach(() => {
    wrapper = shallow(
      <ShoppingCart
        cartItems={cartItems}
        handleRemoveCartItem={handleRemoveCartItemSpy}
      />
    );
  });

  it('should render Shopping Cart correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a list of Cart Items', () => {
    expect(wrapper.find('CartItem')).toHaveLength(cartItems.length);
  });
});
