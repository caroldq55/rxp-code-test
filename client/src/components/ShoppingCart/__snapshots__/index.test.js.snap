// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShoppingCart/> should render Shopping Cart correctly 1`] = `ShallowWrapper {}`;
